src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java

```diff
@@ -98,30 +126,97 @@ public void fillsTestSpecificationIdItems() {
 
     @Test
     public void shouldImport() throws Exception {
-        xltStub.enqueue(new MockResponse()
+        xltestviewMock.enqueue(new MockResponse()
                 .addHeader("Content-Type", "application/json; charset=utf-8")
                 .setBody("{ \"testRunId\": \"testrunid\" }"));
-        FilePath fp = new FilePath(new File("."));
+        FilePath fp = new FilePath(new File(this.getClass().getResource("/demo_test_results").getPath()));
 
         PrintStream logger = new PrintStream(System.out);
 
-        Map<String, Object> metadata = new LinkedHashMap<String, Object>();
-        metadata.put("source", "jenkins");
-        metadata.put("serverUrl", "http://my-jenkins");
-        metadata.put("buildResult", "success");
-        metadata.put("buildNumber", "10");
-        metadata.put("jobUrl", "http://my-jenkins/job/sub-dir/my-job");
-        metadata.put("jobName", "sub-dir/my-job");
-        metadata.put("buildUrl", "http://my-jenkins/job/test/14");
-        metadata.put("executedOn", "slave1");
-        metadata.put("buildParameters", new HashMap());
+        Map<String, Object> metadata = createMetadata();
 
         xlTestServer.uploadTestRun("testspecid", fp, "**/*.xml", null, metadata, logger);
 
-        RecordedRequest request = xltStub.takeRequest();
+        RecordedRequest request = xltestviewMock.takeRequest();
+        verifyUploadRequest(request);
+    }
+
+    @Test
+    public void shouldImportWithTrailingSlash() throws IOException, InterruptedException, MessagingException {
```

Comment:
All in all the test checking the extra URL is very heavy for what it actually tests now. E.g. verifying that XLTestServerImpl.createUrl strips out extra '/'. Which is actually testing a JVM class.